pkg/beacon/relay/node.go

```diff
@@ -157,6 +158,61 @@ func (n *Node) ForwardSignatureShares(groupPublicKeyBytes []byte) {
 	n.netProvider.BroadcastChannelForwarderFor(name)
 }
 
+// ResumeSigningIfEligible enables a client to rejoin the ongoing signing process
+// after it was crashed or restarted and if it belongs to the signing group.
+func (n *Node) ResumeSigningIfEligible(
+	relayChain relayChain.Interface,
+	signing chain.Signing,
+) {
+	isEntryInProgress, err := relayChain.IsEntryInProgress()
+	if err != nil {
+		logger.Errorf(
+			"failed checking if an entry is in progress: [%v]",
+			err,
+		)
+		return
+	}
+
+	if isEntryInProgress {
+		previousEntry, err := relayChain.CurrentRequestPreviousEntry()
+		if err != nil {
+			logger.Errorf(
+				"failed getting previous entry on current request: [%v]",
+				err,
+			)
+			return
+		}
+		entryStartBlock, err := relayChain.CurrentEntryStartBlock()
+		if err != nil {
+			logger.Errorf(
+				"failed getting a current relay entry start block: [%v]",
```

Comment:
`current relay request start block` please and of course we need to rename it in the contract because I failed to do that in my PR, heh.

Reply:
Replaced all occurrences.